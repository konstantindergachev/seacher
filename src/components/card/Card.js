import React from 'react';
import './Card.scss';

const Card = ({ name, img, email }) => {
  return (
    <div className="card__container">
      <img className="card__img" src={img} alt={name} />
      <div>
        <p className="card__name">{name}</p>
        <p className="card__email">{email}</p>
      </div>
    </div>
  );
};

export default Card;
