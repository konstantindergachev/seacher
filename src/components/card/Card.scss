@import '../../scss/mixins.scss';

.card {
  &__container {
    max-width: 250px;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    background-color: #95dada;
    border: 1px solid grey;
    border-radius: 5px;
    padding: 25px;
    cursor: pointer;
    -moz-osx-font-smoothing: grayscale;
    backface-visibility: hidden;
    transform: translateZ(0);
    transition: transform 0.25s ease-out;

    &:hover {
      transform: scale(1.05);
    }

    @include mobfirst(796) {
      margin-bottom: 15px;
    }
    &:not(:nth-child(3n)) {
      @include mobfirst(796) {
        margin-right: 15px;
      }
    }
  }
  &__img {
    margin-bottom: 20px;
  }
  &__name,
  &__email {
    text-align: center;
    font-size: 12px;
  }
}
