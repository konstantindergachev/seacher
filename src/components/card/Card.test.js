import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

describe('Card', () => {
  it('to render Card component', () => {
    expect(shallow(<Card />).length).toEqual(1);
  });

  it('to get the snapshot of Card component', () => {
    const card = renderer.create(<Card />).toJSON();
    expect(card).toMatchSnapshot();
  });
});
