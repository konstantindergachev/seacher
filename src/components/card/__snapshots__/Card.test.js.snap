// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card to get the snapshot of Card component 1`] = `
<div
  className="card__container"
>
  <img
    className="card__img"
  />
  <div>
    <p
      className="card__name"
    />
    <p
      className="card__email"
    />
  </div>
</div>
`;
